File: 05_DataBinding/01_DataBinding/app/src/main/java/com/codingnight/android/databinding/BindingAdapters.java
package com.codingnight.android.databinding;

import android.content.Context;
import android.graphics.drawable.Drawable;

import androidx.core.content.ContextCompat;
import androidx.databinding.BindingAdapter;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class BindingAdapters {

    @BindingAdapter("app:mainFabIcon")
    public static void mainFabIcon(FloatingActionButton button, IncrementMode incrementMode) {
        button.setImageDrawable(getDrawerMainFab(incrementMode, button.getContext()));
    }

    private static Drawable getDrawerMainFab(IncrementMode incrementMode, Context context) {
        switch (incrementMode) {
            case AUTO_TASK_STARTED:
                return ContextCompat.getDrawable(context, R.drawable.ic_pause);
            case AUTO_TASK_STOPED:
                return ContextCompat.getDrawable(context, R.drawable.ic_play);
            default:
                return ContextCompat.getDrawable(context, R.drawable.ic_add);
        }
    }
}

File: 03_ViewModel/03_AndroidViewModel/app/src/main/java/com/codingnight/android/counter/BindingAdapters.kt
package com.codingnight.android.counter

import android.content.Context
import android.graphics.drawable.Drawable
import androidx.core.content.ContextCompat
import androidx.databinding.BindingAdapter
import com.google.android.material.floatingactionbutton.FloatingActionButton

object BindingAdapters {
    @JvmStatic
    @BindingAdapter("app:mainFabIcon")
    fun mainFabIcon(button: FloatingActionButton, incrementMode: IncrementMode) {
        button.setImageDrawable(getDrawerMainFab(incrementMode, button.context))
    }

    private fun getDrawerMainFab(incrementMode: IncrementMode, context: Context): Drawable? {
        return when (incrementMode) {
            IncrementMode.AUTO_TASK_STARTED -> ContextCompat.getDrawable(
                context,
                R.drawable.ic_pause
            )
            IncrementMode.AUTO_TASK_STOPED -> ContextCompat.getDrawable(context, R.drawable.ic_play)
            else -> ContextCompat.getDrawable(context, R.drawable.ic_add)
        }
    }
}